src/site/content/en/blog/ecommerce-notifications-best-practice/index.md

```diff
@@ -0,0 +1,360 @@
+
+Push notifications are great for delivering information about:
+
+*   Blockers to a goal they are trying to achieve. For example:
+
+    *   An item in a customer’s purchase request is out of stock.
+
+    *   A customer’s credit card was declined.
+
+    *   A customer needs to update their shipping address.
+
+*   Time-sensitive actions the customer could take. For example: 
+
+    *  A temporary discount on an item a customer is interested in.
+
+    *   New, limited stock of a popular item that is likely to sell out.
+
+    *   Product pre-releases and early-bird deals.
+
+#### Precise
+
+Notifications should be **short**, **clear**, and **specific**. Tell the
+customer exactly what happened, what you want them to do, or what you’re
+offering.	
+
+#### Actionable
+
+Notifications should provide clear, specific calls-to-action. To maximize the
+likelihood that the customer will take this action, describe it as precisely
+as possible. Make sure that buttons and links clearly indicate what they will
+do or where they will go. 
+
+### Send notifications at the right time
+
+You respect your customers, and their time. Make sure your push notifications
+do too. 
+
+In general, if the information is time-critical, you should notify the customer
+straight away so they don’t miss out. However, you don’t want to send
+notifications so frequently that customers feel frustrated or overwhelmed.
+
+It’s hard to give specific guidelines for how frequently you should send
+notifications to optimize engagement, since each site has unique
+considerations. The goal is to strike a balance so that customers are engaging
+and re-engaging with your site by following through on actions from your push
+notifications.
+
+### Avoid sending obsolete or too-frequent notifications
+
+Frequent notifications will eventually reach a "tipping point", after which
+you’ll see a wave of customers unsubscribing. The less you filter notifications
+for relevance and timeliness, the sooner this tipping point will occur.
+
+The `ttl` (Time To Live) server header controls how long the app will keep
+trying to deliver a notification before abandoning the attempt. Use the `ttl`
+header to avoid sending obsolete notifications.
+
+```js
+const serverOptions = {
+  vapidDetails: {
+    subject: VAPID_SUBJECT,
+    publicKey: VAPID_PUBLIC_KEY,
+    privateKey: VAPID_PRIVATE_KEY,
+  },
+  TTL: notification.ttl,
+};
+
+webpush.sendNotification(subscription, payload, serverOptions);
+```
+
+[See this code in Glitch](https://glitch.com/edit/#!/ecommerce-notifications?path=src/server/routes/test-notifications.js:122:0).
+
+In our sample app, users can also configure the types of notifications they 
+subscribe to:
+
+```html
+<div id="preferences">
+  <p>Send push notifications for:</p>
+  <div class="notification-type">
+    <input id="notifydeals" name="notifydeals" type="checkbox" onChange="window.dispatchEvent(new Event('push-change'))">
```

Comment:
We should avoid inline event handlers, but if we do need to use them, I think they should be all lower case.